raiden/api/python.py

```diff
@@ -700,3 +712,473 @@ def get_channel_events(self, channel_address, from_block, to_block=None):
                 returned_events.append(new_event)
 
         return returned_events
+
+
+class RaidenAPI2:
+    # pylint: disable=too-many-public-methods
+
+    def __init__(self, raiden):
+        self.raiden = raiden
+
+    @property
+    def address(self):
+        return self.raiden.address
+
+    # XXX: This interface will break once the channel identifiers are not addresses
+    def get_channel(self, channel_address):
+        if not isaddress(channel_address):
+            raise InvalidAddress('Expected binary address format for channel in get_channel')
+
+        channel_list = self.get_channel_list()
+        for channel in channel_list:
+            if channel.identifier == channel_address:
+                return channel
+
+        raise ChannelNotFound()
+
+    def token_network_register(self, token_address):
+        """Register the `token_address` in the blockchain.
+
+        Raises:
+            InvalidAddress: If the token_address is not a valid address.
+            ValueError: If the token is already registered.
+            TransactionThrew: If the register transaction failed, this may
+                happen because the account has not enough balance to pay for the
+                gas or this register call raced with another transaction and lost.
+        """
+
+        if not isaddress(token_address):
+            raise InvalidAddress('token_address must be a valid address in binary')
+
+        if token_address in self.get_tokens_list():
+            raise ValueError('Token already registered')
```

Comment:
For the already registered error here perhaps use a specific exception? Since value errors can also be thrown by other things and it would be easier to catch the exception up the stack if it's more specific.